shepherd/py_source_combiner.py

```diff
@@ -0,0 +1,94 @@
+import os
+import re
+import sys
+
+
+def extract_imports(file_content, source_dir):
```

Comment:
You may want to use https://docs.python.org/3/library/ast.html  rather than grep to get the imports, as regexps are  not very robust. For example, I don't think the following case is handled correctly:

```python
if something:
   import ...
```

Reply:
I used this as a temporary combiner for this project. I will remove this file entirely.